packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -822,6 +825,43 @@ class _DraggableScrollableSheetScrollController extends ScrollController {
   }
 }
 
+class _SheetBallisticScrollActivity extends BallisticScrollActivity {
+  _SheetBallisticScrollActivity(
+      _DraggableScrollableSheetScrollPosition scrollPosition,
+      Simulation simulation,
+      {required this.isScrolling})
+      : super(scrollPosition, simulation, scrollPosition.context.vsync, false);
+  @override
+  final bool isScrolling;
+
+  @override
+  _DraggableScrollableSheetScrollPosition get delegate =>
+      super.delegate as _DraggableScrollableSheetScrollPosition;
+
+  @override
+  void applyNewDimensions() {
+    // This is expected to happen continuously while scrolling the outer sheet.
```

Comment:
This causes a regression in the case of an inner scrollable near its end going ballistic. In that case, the outer sheet expands but now doesn't update the inner scrollable until it's finished.

This change complicates this case since the outer sheet scroll is now its on activity, which cannot coexist with the inner scrollable ballistic scroll that would normally handle the resize.

Reply:
Turns out this wasn't the cause of the regression; it was because the velocity of this activity was nonzero. Since the inner content shouldn't be moving (relatively speaking-ish), setting the velocity to zero is fine and lets the physics handle the range maintenance.

https://api.flutter.dev/flutter/widgets/RangeMaintainingScrollPhysics-class.html